expression/builtin_cast_vec.go

```diff
@@ -583,11 +582,36 @@ func (b *builtinCastTimeAsDurationSig) vecEvalDuration(input *chunk.Chunk, resul
 }
 
 func (b *builtinCastDurationAsDurationSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDurationAsDurationSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	dec := b.tp.Decimal
+	var dur types.Duration
+	result.ResizeGoDuration(n, false)
+	result.MergeNulls(buf)
+	res := result.GoDurations()
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			continue
+		}
+		dur, err = buf.GetDuration(i, -1).RoundFrac(int8(dec))
```

Comment:
Please use the vectorized method to get the argument, see https://godoc.org/github.com/pingcap/tidb/util/chunk#Column.GoDurations

Reply:
thank you, i will try it ~